File: Hand.py
from Card import Card

class Hand(object):
	
	def __init__(self):
		self.handValue = 0
		self.ace = False
		self.hand = []


	def addCard(self, card):
		self.hand.append(card)

	def getCardCount(self):
		return len(self.hand)

	def getCard(self, position):
		return self.hand[position]


	def getHandValue(self):
		self.cards = self.getCardCount()

		for i in range(self.cards):
			self.card = self.getCard(i)
			self.cardValue = self.card.getValue()

			if self.cardValue == 1:
				self.ace = True
			self.handValue = self.handValue + self.cardValue

			if self.ace and (self.handValue + 10) <= 21:
				self.handValue = self.handValue + 10
		return self.handValue

	def __str__(self):
		return '\n'.join(map(str, self.hand))

File: Hand.java
package Blackjack;

import java.util.ArrayList;

public class Hand {

    private ArrayList<Card> hand;

    public Hand() {
	int val, cards;
	boolean ace;
        hand = new ArrayList<Card>();
    }

    public void addCard(Card card) {
            hand.add(card);
    }

    public int getCardCount() {
        return hand.size();
    }

    public Card getCard(int position) {
            return (Card)hand.get(position);
    }



    public int getHandValue() {
        val = 0;
        ace = false;
        cards = getCardCount();

        for(int i = 0;  i < cards;  i++) {
            Card card;
            int cardVal;
            card = getCard(i);
            cardVal = card.getValue();
            
	    if(cardVal > 10) {
                cardVal = 10;
            }
            
	    if(cardVal == 1) {
                ace = true;
            }
            val = val + cardVal;
        }


        if (ace &&  (val + 10) <= 21 )
            val = val + 10;

        return val;

    }

}
